File: src/main/java/com/nir/beans/HardcodedMethodComponent.java
package com.nir.beans;

import com.google.common.collect.Lists;
import com.nir.utils.math.method.Method;
import com.nir.utils.math.method.hardcoded.ForwardEuler;
import com.nir.utils.math.method.hardcoded.RungeKutta;

import java.util.List;

public class HardcodedMethodComponent {
    public List<Method> getAll() {

        return Lists.newArrayList(new RungeKutta(4), new RungeKutta(5), ForwardEuler.INSTANCE);
    }
}

File: src/main/kotlin/com/nir/beans/method/HardcodedMethodComponent.kt
package com.nir.beans.method

import com.nir.beans.method.hardcoded.ForwardEuler
import com.nir.beans.method.hardcoded.RungeKutta

class HardcodedMethodComponent {
    val getMethods: List<Method>
        get() = arrayListOf(RungeKutta(4), RungeKutta(5), ForwardEuler)
}